test/new-e2e/pkg/utils/e2e/client/vm.go

```diff
@@ -21,11 +22,13 @@ var _ clientService[commonvm.ClientData] = (*VM)(nil)
 type VM struct {
 	*UpResultDeserializer[commonvm.ClientData]
 	*vmClient
+	OS commonos.OS
```

Comment:
I don't think `OS` should be exported as most methods doesn't make sense for `client.VM`. If you look at  [Agent code](https://github.com/DataDog/datadog-agent/blob/972c3b86fd2bffbf615b31b55db2f5fffba6879a/test/new-e2e/pkg/utils/e2e/client/agent.go#L26) `os` is not exported.